eth/fetcher/block_fetcher.go

```diff
@@ -359,13 +359,7 @@ func (f *BlockFetcher) loop() {
 			}
 			// If too high up the chain or phase, continue later
 			number := op.number()
-			if number > height+1 {
```

Comment:
Doesn't this prevent us from having to cache a potentially infinite number of blocks during download? Pretty sure this check (or some variation of it) still needs to exist.